### PR TITLE
Cod 2/cod 6

### PR DESCRIPTION
**Pull request by eldarht**

### CODE REVIEW

File: apiServer/model/mongodb.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	// Set up currency collation indexing
+	index := mgo.Index{
+		Key:        []string{"uri"},
+		Unique:     true,
+		DropDups:   true,
+		Background: true,
+		Sparse:     true,
+	}
+
+	if err != nil {
```
Comment (Xillez) [inline apiServer/model/mongodb.go]: This if is not needed, checked above

File: apiServer/model/mongodb.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+// It returns empty repo if it is not in db.  
+func (db *MongoDB) findRepoByURI(uri string) (repo RepoModel, err error) {
+	session, err := mgo.Dial(db.DatabaseURL)
+	if err != nil {
+		log.Fatalln(logError+"Can't connect to database: ", err)
+	}
+	defer session.Close()
+
+	// Retrun empty repo with error if error is not "Not found"
+	if err = session.DB(db.DatabaseName).C(db.RepoColl).Find(bson.M{"uri": uri}).One(&repo); err != nil && err.Error() != "not found" {
```
Comment (Xillez) [inline apiServer/model/mongodb.go]: Doing the thing you wanna do while checking it, is cool, but is kinda hard to read. If you wanna keep it, do so.

Comment (Xillez) [reply apiServer/model/mongodb.go]: Especially these long expressions.

**Comment by zohaib194** [comment]
> apiServer/apiServer is a binary file, is it needed?

**Pull request merged**